src/main/java/org/hobbit/core/components/AbstractTaskGenerator.java

```diff
@@ -61,6 +62,8 @@
      * Mutex used to wait for the start signal after the component has been started
      * and initialized.
      */
+    final private Semaphore currentlyProcessedMessages = new Semaphore(0);
```

Comment:
Inserting the new attribute here is not a good idea – simply because it has the javadoc comment of the `startTaskGenMutex` attribute. It should have its own javadoc comment and all other attributes should of course keep their javadoc comments.